### PR TITLE
Implement minimized action to hide the application on close

### PR DESCRIPTION
**Pull request by Warkanlock**
This will minimize the application instead of closing it when the user hits the close button

Set **minimize on close** to default always, but when is enabled user cannot shut down the process using the window close button.

image

Related to issue #59

### CODE REVIEW

File: src/scripts/tray.js [hunk 1]
```diff
@@ -1,22 +1,48 @@
...
-    // do nothing on click
+    if (mainWindow) {
+      mainWindow.show();
+    }
   });
 
   tray.setToolTip("Tidal-hifi");
-  tray.setContextMenu(getMenu());
+  
+  if (mainWindow && store.get(settings.minimizeOnClose)) {
```
Comment (Mastermindzh) [inline src/scripts/tray.js:25]: We lose the entire menu like this, we should do either one of two solutions:

- Add the items to the menu regardless of the setting and just use `getMenu()`
- Merge the result of `getMenu()` with this new menu.

Either way is fine for me. Being able to focus (show) and close the app from the menu is nice anyway.

**Review by Mastermindzh — CHANGES_REQUESTED** [review]
Looks very good! Just the menu thing that I don't really agree with.

File: src/scripts/tray.js (hunk 1 above)
Comment (Warkanlock) [reply src/scripts/tray.js:25]: Yeah, I did that (I left a https://github.com/Mastermindzh/tidal-hifi/issues/59#issuecomment-862293724 on the issue thread) but I didn't know if you were about to agree with it. If the comment looks OK I'll implement it

**Comment by Mastermindzh** [comment]
Ok, here's the deal. It's bloody hot where I live so I'm going to take the night off.
I'll test this change tomorrow (during work, I like music during work...) and if all goes well I'll merge and release it tomorrow evening. 

Thanks for the contribution, great work!

**Review by Mar0xy — COMMENTED** [review]
This looks great but I feel like Open App and Close App should be merged to one button called something like **Toggle Window** cause I feel like having both options + all the other options makes the Menu clustered.

**Comment by Warkanlock** [comment]
> This looks great but I feel like the Open App button should also include the ability to minimize it to the tray and be renamed to **Toggle Window** in case you just wanna quickly check something and don't want to move the cursor back over to the close button. As for the Close App button, I think it should be renamed to something like **Exit (app name)** or just **Quit**

Makes a lot of sense to me, if @Mastermindzh agree I can implement it

**Review by Mar0xy — APPROVED** [review]
I think this is good this way only need to wait on @Mastermindzh opinion.

**Comment by Mastermindzh** [comment]
Hehe... Know what's funny? I've been on the i3 window manager for a good few years now but I forgot that I can't actually minimize applications 🤣 .

Had to spin up a VM again.

Anyway, let's get to work!

**Pull request merged**